### PR TITLE
feat(api): update my classroom mode

### LINKED ISSUE #51793: PUT /settings/update-my-classroom-mode
See https://github.com/freeCodeCamp/freeCodeCamp/pull/51708

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #51793

### CODE REVIEW

File: api/prisma/oldschema.prisma [hunk 1]
```diff
@@ -198,6 +198,7 @@ model user {
   verificationToken            String? // Undefined
   website                      String? // Undefined
   yearsTopContributor          String[]             @default([]) // Undefined | String[]
+  isClassroomAccount           Boolean? // Undefined
```
Comment (Sembauke) [inline api/prisma/oldschema.prisma:201]: I was not sure if I should add this to the old schema as well.

File: api/src/routes/settings.test.ts [hunk 2]
```diff
@@ -689,6 +689,29 @@ Please wait 5 minutes to resend an authentication link.`
...
+        });
+
+        expect(response.body).toEqual({
+          message: 'flash.classroom-mode-updated',
+          type: 'success'
+        });
+        expect(response.statusCode).toEqual(200);
+      });
+
+      test('PUT returns 400 status code with invalid classroomMode', async () => {
```
Comment (moT01) [inline api/src/routes/settings.test.ts:706]: Not sure if this really matters, but the current API sends a 403 back when there isn't a valid request body.

File: api/src/routes/settings.test.ts [hunk 3]
```diff
@@ -689,6 +689,29 @@ Please wait 5 minutes to resend an authentication link.`
         expect(response.statusCode).toEqual(400);
       });
     });
+
+    describe('/update-my-classroom-mode', () => {
+      test('PUT returns 200 status code with "success" message', async () => {
```
Comment (moT01) [inline api/src/routes/settings.test.ts:694]: We could maybe test that the database gets updated after this request.

**Review by moT01 — COMMENTED** [review]
Looking good @Sembauke 🎉 

Left a few comments. Not sure if any of it really needs to be changed.

File: api/prisma/oldschema.prisma (hunk 1 above)
Comment (moT01) [reply api/prisma/oldschema.prisma:201]: I don't see a probably with adding this.

**Review by moT01 — APPROVED** [review]
LGTM 🎉

File: api/src/schemas.ts [hunk 4]
```diff
@@ -293,6 +293,21 @@ export const schemas = {
...
+  updateMyClassroomMode: {
+    body: Type.Object({
+      isClassroomAccount: Type.Boolean()
+    }),
+    response: {
+      200: Type.Object({
+        message: Type.Literal('flash.classroom-mode-updated'),
+        type: Type.Literal('success')
+      }),
+      400: Type.Object({
```
Comment (ShaunSHamilton) [inline api/src/schemas.ts:305]: In the route, this is defined as a `500`.

Comment (ShaunSHamilton) [reply api/src/schemas.ts:305]: In the old api, this is defined as a `403`.

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]

File: api/src/schemas.ts (hunk 4 above)
Comment (Sembauke) [reply api/src/schemas.ts:305]: I guess we want everything to be the same then?

**Review by ShaunSHamilton — APPROVED** [review]
Thank you. LgTm 👍

**Pull request merged**